File: batch/src/main/java/com/beat/batch/promotion/job/PromotionMaintenanceJob.java
package com.beat.batch.promotion.job;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import com.beat.batch.promotion.facade.PromotionMaintenanceFacade;

import lombok.RequiredArgsConstructor;

@Component
@RequiredArgsConstructor
public class PromotionMaintenanceJob {

	private final PromotionMaintenanceFacade promotionMaintenanceFacade;

	@Value("${beat.scheduler.owner:false}")
	private boolean schedulerOwner;

	@Scheduled(cron = "1 0 0 * * ?", scheduler = "maintenanceTaskScheduler")
	public void checkAndDeleteInvalidPromotions() {
		if (!schedulerOwner) {
			return;
		}

		promotionMaintenanceFacade.checkAndDeleteInvalidPromotions();
	}
}

File: batch/src/main/kotlin/com/beat/batch/promotion/job/PromotionMaintenanceJob.kt
package com.beat.batch.promotion.job

import com.beat.application.system.promotion.command.PromotionMaintenanceService
import org.springframework.beans.factory.annotation.Value
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component

@Component
class PromotionMaintenanceJob(
    private val promotionMaintenanceService: PromotionMaintenanceService,
    @param:Value("\${beat.scheduler.owner:false}") private val schedulerOwner: Boolean,
) {
    @Scheduled(cron = "1 0 0 * * ?", scheduler = "maintenanceTaskScheduler")
    fun checkAndDeleteInvalidPromotions() {
        if (!schedulerOwner) return
        promotionMaintenanceService.checkAndDeleteInvalidPromotions()
    }
}
